ftdc/ftdc.go

```diff
@@ -94,6 +96,10 @@ type FTDC struct {
 	// detailed description.
 	prevFlatData []float32
 
+	readStatsWorker  *utils.StoppableWorkers
```

Comment:
So my first use of the ticket case I added a bit ago was....one ticker worker (reading stats) and one non-ticker worker (writing stats).

Which we explicitly didn't set out to support. But now I have an idea of the requirements. The non-ticker worker is consuming from a channel written to by the ticker worker.

Reply:
IIRC as long as we don't have > 1 ticker worker, you should be good.

Reply:
I think we can add it, but I'm not positive that means I can remove the `outputWorkerDone chan struct{}` variable (which is a non-trivial line of reasoning -- the margin is too small for this proof).

And if I can't use the StoppableWorkers to _completely_ control shutdown, I'd prefer to stay on track of getting FTDC deployed.

FWIW, I think we can add to StoppableWorkers to make this possible, but I've already done enough speculative patch pushing that was only 90% usable.

Reply:
I would buy that `StoppableWorkers` is not enough here and you specifically need synchronization in the form of a done channel for the output worker. What you have seems good, and I wouldn't want to "force" the other concurrency primitives into a `StoppableWorkers`.